src/xarray_regrid/utils.py

```diff
@@ -388,3 +369,50 @@ def coord_is_covered(
     left_covered = obj[coord].min() <= target[coord].min() - pad
     right_covered = obj[coord].max() >= target[coord].max() + pad
     return bool(left_covered.item() and right_covered.item())
+
+
+@overload
+def ensure_monotonic(obj: xr.DataArray, coord: Hashable) -> xr.DataArray: ...
+
+
+@overload
+def ensure_monotonic(obj: xr.Dataset, coord: Hashable) -> xr.Dataset: ...
+
+
+def ensure_monotonic(
```

Comment:
I was also working on this exact function for #46, see https://github.com/xarray-contrib/xarray-regrid/pull/46#discussion_r1751955505.

You can use the following to make the code more simple:
```python
obj.indexes[coord].is_monotonic_increasing
obj.indexes[coord].is_unique
```

I'm also not sure what is more efficient: sorting/dropping duplicates one coord at a time, or tackling all in one go. My approach was this;
```py
def ensure_monotonic(
    obj: xr.Dataset | xr.DataArray, coords: list[Hashable]
) -> xr.Dataset | xr.DataArray:
    """Sort coordinates that do not monotonically increase.

    If no sorting needs to take place, the data is not sorted."""
    unsorted_coords = [
        coord for coord in coords if not obj.indexes[coord].is_monotonic_increasing
    ]
    if len(unsorted_coords) > 0:
        obj = obj.sortby(unsorted_coords)

    duplicate_coords = [coord for coord in coords if not obj.indexes[coord].is_unique]
    if len(duplicate_coords) > 0:
        obj = obj.drop_duplicates(duplicate_coords)

    return obj
```